### DISCUSSION TITLE
Multiple Azurerm providers supported?

### DISCUSSION THREAD

**Discussion by rbmiller12** [gettek/terraform-azurerm-policy-as-code#76] [Q&A]
Are multiple azurerm providers supported (so I can reference different subscriptions using alias)?
Thanks!

I need to reference a specific resource group for an exemption (using data azurerm_resource_group) that is in a different subscription (we have several, so this will be a common need).

```hcl
provider "azurerm" {
  skip_provider_registration = true
  subscription_id = "xxxxxxxxxxx"
  features {}
}
provider "azurerm" {
  skip_provider_registration = true
  subscription_id = "yyyyyyyyyy"
  features {}
  alias = "dev"
}


data azurerm_resource_group devopenairg {
  provider = azurerm.dev
  name = "devrg"
}


module "exemption_rg_dev_openai_region" {
  source               = "gettek/policy-as-code/azurerm//modules/exemption"
  name                 = "devopenai_region_exemption"
  display_name         = "Exempted due to OpenAI availability"
  description          = "Excludes Corp OpenAI from region restriction due to availability"
  scope                = data.azurerm_resource_group.devopenairg.id
  policy_assignment_id = module.org_mg_general_governance_initiative.id

  # use member_definition_names for simplicity when policy_definition_reference_ids are unknown
  member_definition_names = [
    "allowed_locations"
  ]
}
```

**Comment by gettek** [comment]
Hi @rbmiller12 that's fine as long as all resources and resource groups sit under the policy assignment scope. If you collate all your resource group `data.rg.id` outputs into a `local {}` you can loop around them like in [this example](https://github.com/gettek/terraform-azurerm-policy-as-code/tree/main/modules/exemption#multiple-resource-exemptions-for-all-keyvaults-in-a-resource-group)

**Reply by rbmiller12** [reply]
My issue with multiple providers pointing to different subscriptions has been resolved.  It is working as expected now.
